bodo/pandas/_executor.cpp

```diff
@@ -9,20 +9,58 @@
 #include "../libs/streaming/_shuffle.h"
 #include "_plan.h"
 #include "arrow/io/api.h"
+#include "duckdb/planner/expression/bound_columnref_expression.hpp"
 #include "parquet/arrow/reader.h"
 
 Executor::Executor(std::unique_ptr<duckdb::LogicalOperator> plan) {
+    processNode(plan);
+}
+
+std::shared_ptr<PhysicalOperator> Executor::processNode(
+    std::unique_ptr<duckdb::LogicalOperator>& plan) {
     // Convert logical plan to physical plan and create query pipelines
 
-    // TODO: support nodes other than read parquet
-    duckdb::LogicalGet& get_plan = plan->Cast<duckdb::LogicalGet>();
+    // TODO: support all node types
+    duckdb::LogicalOperatorType ptype = plan.get()->type;
+    if (ptype == duckdb::LogicalOperatorType::LOGICAL_GET) {
+        duckdb::LogicalGet& get_plan = plan->Cast<duckdb::LogicalGet>();
+
+        std::shared_ptr<PhysicalOperator> physical_op =
+            get_plan.bind_data->Cast<BodoScanFunctionData>()
+                .CreatePhysicalOperator();
+
+        pipelines.emplace_back(
+            std::vector<std::shared_ptr<PhysicalOperator>>({physical_op}));
+        return physical_op;
+    } else if (ptype == duckdb::LogicalOperatorType::LOGICAL_PROJECTION) {
+        return PhysicalProjection::make(plan->Cast<duckdb::LogicalProjection>(),
+                                        this);
+    } else {
+        throw std::runtime_error(
+            std::string("Executor doesn't handle logical operator of type ") +
+            std::to_string(static_cast<int>(ptype)));
+    }
+}
 
+std::shared_ptr<PhysicalOperator> PhysicalProjection::make(
+    duckdb::LogicalProjection& proj_plan, Executor* executor) {
+    // Process the source of this projection.
+    std::shared_ptr<PhysicalOperator> source =
+        executor->processNode(proj_plan.children[0]);
+    std::vector<int> selected_columns;
+    // Convert BoundColumnRefExpressions in LogicalOperator.expresssions field
+    // to integer selected columns.
+    for (const auto& expr : proj_plan.expressions) {
+        duckdb::BoundColumnRefExpression& colref =
+            expr->Cast<duckdb::BoundColumnRefExpression>();
+        selected_columns.push_back(colref.binding.column_index);
+    }
```

Comment:
Did we figure out the output column index vs. input column index problem I saw previously?

Reply:
I don't think we've bottomed out on this issue.